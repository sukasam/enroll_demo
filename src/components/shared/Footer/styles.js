import { css } from "@emotion/react";

export default css`
    color: #153862;
    font-family: "Inter", Arial, Helvetica, sans-serif;

    .bolder {
        font-weight: 600;
        margin-top: 0;
        color: #153862;
    }
    .logo {
        margin-bottom: 10px;
        @media (max-width: 600px) {
            display: flex;
            align-items: center;
            height: 26px;
            margin: 0 auto 20px auto;
        }
    }
    .social-link {
        color: #153862;
        @media (max-width: 600px) {
            margin: none;
        }
    }
    .links {
        text-decoration: none;
        margin-bottom: 10px;
        color: #153862;
        display: flex;
        width: max-content;

        &:hover {
            color: #99bcdf;
            text-decoration: underline;
        }
    }
    .bottom {
        background: #0b3b61;
        color: #ffffff;
    }

    .top-wrapper {
        border-top: 5px solid #99bcdf;
        display: flex;
        width: 100%;
    }

    .top {
        display: flex;
        margin: auto;
        max-width: calc(1170px - 48px);
        width: 100%;
        padding: 60px 40px 40px;
        flex-direction: column;

        @media (min-width: 900px) {
            flex-direction: row;
        }

        .social,
        .section {
            flex: 1;
        }

        .social {
            display: flex;
            flex-direction: column;
            justify-content: center;
            max-width: 330px;
            margin: auto;
            width: 100%;

            @media (min-width: 900px) {
                margin-left: 0;
                margin-top: 0;
                justify-content: flex-start;
            }

            .social-links {
                width: 100%;
                display: flex;
                justify-content: space-between;
                max-width: 280px;
                margin: auto;
                margin-top: 20px;
                margin-bottom: 50px;

                @media (min-width: 900px) {
                    margin-left: 0px;
                    max-width: 200px;
                }
            }

            .logo {
                height: 26px;
                width: max-content;
                margin: auto;

                @media (min-width: 900px) {
                    margin-left: 0;
                }
            }
        }

        .sections {
            display: flex;
            flex-direction: column;
            flex: 1;

            @media (min-width: 560px) {
                flex-direction: row;
            }

            .section {
                margin-bottom: 24px;

                @media (min-width: 560px) {
                    width: 50%;
                }
            }
        }
    }

    .bottom-items {
        width: 100%;
        padding: 24px;
        margin: auto;
        display: flex;
        flex-direction: column;

        .policy-container {
            display: flex;
            height: 100%;
            align-items: center;
            font-size: 12px;
            margin-left: 80px;

            @media (max-width: 900px) {
                width: auto;
                margin-left: 0;
                display: flex;
                justify-content: center;
            }
        }

        .copyright {
            display: flex;
            height: 100%;
            align-items: center;
            font-size: 10px;
            margin: auto;
            width: max-content;
            max-width: 100%;
            text-align: center;
            margin-bottom: 10px;

            @media (min-width: 560px) {
                text-align: left;
                margin: 0;
                margin-right: auto;
            }

            @media (min-width: 900px) {
                font-size: 12px;
                margin: 0;
            }
        }

        .separator {
            margin: 0 10px;
        }

        @media (min-width: 900px) {
            text-align: left;
            flex-direction: row;
            width: calc(1130px + 48px);
        }

        @media (min-width: 560px) {
            text-align: left;
            flex-direction: row;
        }
    }
    .policy {
        color: #ffffff;
        text-decoration: underline;
        &:hover {
            cursor: pointer;

            color: #99bcdf;
        }
    }
    .tou {
        color: #ffffff;

        text-decoration: underline;
        &:hover {
            cursor: pointer;
            color: #99bcdf;
        }
    }
`;
