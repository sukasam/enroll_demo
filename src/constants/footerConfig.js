const footerConfig = {
    socialLinks: [
        {
            key: "footer_facebook_link",
            icon: "/svg/facebook.svg",
            alt: "footer_facebook"
        },
        {
            key: "footer_twitter_link",
            icon: "/svg/x.svg",
            alt: "footer_twitter"
        },
        {
            key: "footer_pintrest_link",
            icon: "/svg/pintrest.svg",
            alt: "footer_pintrest"
        },
        {
            key: "footer_instagram_link",
            icon: "/svg/instagram.svg",
            alt: "footer_instagram"
        },
        {
            key: "footer_youtube_link",
            icon: "/svg/youtube.svg",
            alt: "footer_youtube"
        }
    ],
    sections: [
        {
            title: "footer_about",
            links: [
                {
                    key: "footer_our_company_link",
                    text: "footer_our_company"
                },
                {
                    key: "footer_science_link",
                    text: "footer_science"
                },
                {
                    key: "footer_unicity_blog_link",
                    text: "footer_unicity_blog"
                }
            ]
        },
        {
            title: "footer_customer_support",
            links: [
                {
                    key: "footer_contact_us_link",
                    text: "footer_contact_us"
                }
            ]
        }
    ],
    policies: [
        {
            key: "footer_privacy_policy_link",
            text: "footer_privacy_policy"
        },
        {
            key: "footer_terms_of_use_link",
            text: "footer_terms_of_use"
        }
    ]
};

export default footerConfig;
