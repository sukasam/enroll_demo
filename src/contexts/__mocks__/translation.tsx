import { Alpha2 } from "Constants/countryConfig/enums.js";

export const useTranslations = jest.fn().mockReturnValue({
    country: "US" as Alpha2,
    language: "en",
    setCountry: jest.fn(),
    setLanguage: jest.fn(),
    fetchTranslations: jest.fn(),
    translations: {}
});

export default useTranslations;
