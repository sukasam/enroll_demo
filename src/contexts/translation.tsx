import { debounce } from "@mui/material";
import { translationsSvc } from "Hydra/translationsSvc";
import { sendEvent } from "Services/googleAnalytics";
import { getCookie, setCookie } from "cookies-next";
import {
    ReactNode,
    createContext,
    useCallback,
    useContext,
    useEffect,
    useMemo,
    useState
} from "react";

export interface TranslationContextType {
    translations: Record<string, string>;
    basicTranslations: Record<string, string>;
    country: string;
    language: string;
    setTranslations: (translations: Record<string, string>) => void;
    setBasicTranslations: (basicTranslations: Record<string, string>) => void;
    setCountry: (country: string) => void;
    setLanguage: (language: string) => void;
    fetchTranslations: (country: string, language: string) => void;
}

const defaultState: TranslationContextType = {
    translations: {},
    basicTranslations: {},
    country: getCookie("country") || "US",
    language: getCookie("language") || "en",
    setTranslations: () => null,
    setBasicTranslations: () => null,
    setCountry: () => null,
    setLanguage: () => null,
    fetchTranslations: () => null
};

function getInitialState(): TranslationContextType {
    if (typeof window !== "undefined") {
        const cachedData = sessionStorage.getItem("translationContext");
        if (cachedData) {
            try {
                return JSON.parse(cachedData);
            } catch (error) {
                console.error(
                    "Failed to parse translation context data from sessionStorage:",
                    error
                );
            }
        }
    }
    return defaultState;
}

const TranslationContext = createContext<TranslationContextType>(defaultState);
TranslationContext.displayName = "Translations Context";

function HtmlLangSync(props: { language: string }): React.ReactElement | null {
    const { language } = props;
    useEffect(() => {
        if (typeof document !== "undefined" && language) {
            document.documentElement.lang = language;
        }
    }, [language]);
    return null;
}

export function TranslationProvider({
    children,
    testValues
}: {
    children: ReactNode;
    testValues?: Record<string, string>;
}): React.ReactElement {
    const initialState = useMemo(() => getInitialState(), []);

    const [translations, setTranslations] = useState<Record<string, string>>(
        initialState.translations
    );
    const [basicTranslations, setBasicTranslations] = useState<
        Record<string, string>
    >(initialState.basicTranslations);
    const [country, setCountry] = useState<string>(initialState.country);
    const [language, setLanguage] = useState<string>(initialState.language);

    useEffect(() => {
        setCookie("country", country);
        setCookie("language", language);
        sendEvent("page_configration", {
            country,
            language
        });
    }, [country, language]);

    const fetchTranslations = debounce(
        async (_country: string, _language: string) => {
            try {
                const response = await translationsSvc.get({
                    country: _country || country,
                    language: _language || language
                });

                if (response.ok === false) {
                    throw new Error("Failed to fetch translations");
                }
                // const translations = await response.json();
                setTranslations(response);
            } catch (error) {
                console.error("Error fetching translations:", error);
            }
        },
        500
    );

    const updateCachedTranslationData = useCallback(() => {
        if (typeof window !== "undefined") {
            const dataToCache = {
                translations,
                basicTranslations,
                country,
                language
            };
            try {
                sessionStorage.setItem(
                    "translationContext",
                    JSON.stringify(dataToCache)
                );
            } catch (error) {
                console.error(
                    "Failed to save translation context data to sessionStorage:",
                    error
                );
            }
        }
    }, [translations, basicTranslations, country, language]);

    useEffect(() => {
        updateCachedTranslationData();
    }, [updateCachedTranslationData]);

    const value = useMemo(
        () => ({
            translations,
            basicTranslations,
            country,
            language,
            setTranslations,
            setBasicTranslations,
            setCountry,
            setLanguage,
            fetchTranslations
        }),
        [translations, basicTranslations, country, language, fetchTranslations]
    );

    const providerValue = useMemo(
        () => ({
            ...value,
            ...testValues
        }),
        [value, testValues]
    );

    return (
        <TranslationContext.Provider value={providerValue}>
            <HtmlLangSync language={language} />
            {children}
        </TranslationContext.Provider>
    );
}

export const useTranslations = (): TranslationContextType =>
    useContext(TranslationContext);
