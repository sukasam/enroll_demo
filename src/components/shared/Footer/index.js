/** @jsxImportSource @emotion/react */
import { Grid } from "@mui/material";
import { useHasTranslation, useTranslate } from "Components/shared/Translate";
import footerConfig from "Constants/footerConfig";
import Image from "next/image";
import Link from "next/link";
import { useEffect, useState } from "react";
import mixpanelService from "Services/mixpanel/initializeMixPanel";
import MixpanelEvent from "Services/mixpanel/mixpanelEvent";
import clsx from "Services/utils/clsx";
import BranchName from "./components/BranchName";
import styles from "./styles";

const ANALYTICS_MAPPING = {
    footer_contact_us_link: "contact",
    footer_our_company_link: "our_company",
    footer_science_link: "science",
    footer_unicity_blog_link: "unicity_blog",
    footer_twitter_link: "social_x",
    footer_facebook_link: "social_facebook",
    footer_pintrest_link: "social_pinterest",
    footer_instagram_link: "social_instagram",
    footer_youtube_link: "social_youtube"
};

const trackLinkClick = key => {
    const actionType = ANALYTICS_MAPPING[key];
    if (actionType) {
        try {
            mixpanelService.trackEvent(MixpanelEvent.FOOTER_LINK_CLICKED, {
                action: actionType
            });
        } catch (error) {
            console.error("Failed to track footer link click:", error);
        }
    }
};

const Footer = () => {
    const translate = useTranslate();
    const hasTranslation = useHasTranslation();
    const [isClient, setIsClient] = useState(false);

    useEffect(() => {
        setIsClient(true);
    }, []);

    const dynamicContent = isClient ? (
        <footer className="footer" css={styles}>
            <div className="top-wrapper">
                <div className="top">
                    <div className="social">
                        <Image
                            alt="Unicity"
                            className="logo"
                            src="/svg/logo.svg"
                            width={140}
                            height={25}
                            data-testid="Unicity Logo"
                        />
                        <div className="social-links">
                            {footerConfig.socialLinks.map(link =>
                                hasTranslation(link.key) ? (
                                    <Link
                                        key={link.key}
                                        className="social-link"
                                        href={translate(link.key)}
                                        rel="noreferrer"
                                        target="_blank"
                                        data-testid={link.key}
                                        onClick={() => trackLinkClick(link.key)}
                                    >
                                        <Image
                                            alt={translate(link.alt)}
                                            src={link.icon}
                                            width={24}
                                            height={24}
                                        />
                                    </Link>
                                ) : null
                            )}
                        </div>
                    </div>

                    <div className="sections">
                        {footerConfig.sections.map(section => (
                            <div className="section" key={section.title}>
                                <p
                                    className="bolder"
                                    data-testid={section.title}
                                >
                                    {translate(section.title)}
                                </p>
                                {section.links.map(link =>
                                    hasTranslation(link.key) ? (
                                        <Link
                                            key={link.key}
                                            className="links"
                                            href={translate(link.key)}
                                            rel="noreferrer"
                                            target="_blank"
                                            data-testid={link.key}
                                            onClick={() =>
                                                trackLinkClick(link.key)
                                            }
                                        >
                                            <div>{translate(link.text)}</div>
                                        </Link>
                                    ) : null
                                )}
                            </div>
                        ))}
                    </div>
                </div>
            </div>
            <Grid className="bottom" container>
                <div className={clsx("bottom-items")}>
                    <div className="copyright" data-testid="copyright_label">
                        {translate("footer_copyright").replace(
                            "{{year}}",
                            new Date().getFullYear()
                        )}
                    </div>
                    <div className={clsx("policy-container")}>
                        {footerConfig.policies.map((policy, index) => (
                            <div key={policy.key}>
                                <Link
                                    className="policy"
                                    href={translate(policy.key)}
                                    rel="noreferrer"
                                    target="_blank"
                                    data-testid={policy.key}
                                >
                                    {translate(policy.text)}
                                </Link>
                                {index < footerConfig.policies.length - 1 && (
                                    <span className="separator">|</span>
                                )}
                            </div>
                        ))}
                    </div>
                </div>
                <BranchName />
            </Grid>
        </footer>
    ) : null;

    return dynamicContent;
};

export default Footer;
