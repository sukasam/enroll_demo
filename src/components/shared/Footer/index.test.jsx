/** @jsxImportSource @emotion/react */
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";
import footerConfig from "Constants/footerConfig";
import { TranslationProvider } from "Contexts/translation";
import mixpanelService from "Services/mixpanel/initializeMixPanel";
import Footer from "./index";

jest.mock("next/image", () => props => <img {...props} alt="test-img" />);
jest.mock("./components/BranchName", () => ({
    __esModule: true,
    default: () => <div data-testid="branch-name">BranchName</div>
}));
jest.mock("Services/mixpanel/initializeMixPanel", () => ({
    trackEvent: jest.fn()
}));

const setup = (translationProps = {}) =>
    render(
        <TranslationProvider
            testValues={{
                translations: {
                    unicity_logo: "Unicity Logo",
                    footer_copyright:
                        "© 2024 Copyright Unicity International, Inc. All Rights Reserved.",
                    ...footerConfig.socialLinks.reduce((acc, link) => {
                        acc[link.key] = link.key;
                        acc[link.alt] = link.alt;
                        return acc;
                    }, {}),
                    ...footerConfig.sections.reduce((acc, section) => {
                        acc[section.title] = section.title;
                        section.links.forEach(link => {
                            acc[link.key] = link.key;
                            acc[link.text] = link.text;
                        });
                        return acc;
                    }, {}),
                    ...footerConfig.policies.reduce((acc, policy) => {
                        acc[policy.key] = policy.text;
                        return acc;
                    }, {}),
                    ...translationProps.translations
                },
                country: "US",
                language: "en",
                ...translationProps
            }}
        >
            <Footer />
        </TranslationProvider>
    );

describe("Footer component", () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    test("matches snapshot", () => {
        const { container } = setup();
        expect(container.firstChild).toMatchSnapshot();
    });

    test("renders the Unicity logo", () => {
        const { getByTestId } = setup();
        const logo = getByTestId("Unicity Logo");
        expect(logo).toBeInTheDocument();
    });

    test("renders social links", () => {
        const { getByTestId } = setup();
        footerConfig.socialLinks.forEach(link => {
            const socialLink = getByTestId(link.key);
            expect(socialLink).toBeInTheDocument();
        });
    });

    test("renders section titles", () => {
        const { getByTestId } = setup();
        footerConfig.sections.forEach(section => {
            const sectionTitle = getByTestId(section.title);
            expect(sectionTitle).toBeInTheDocument();
        });
    });

    test("renders policy links", () => {
        const { getByTestId } = setup();
        footerConfig.policies.forEach(policy => {
            const policyLink = getByTestId(policy.key);
            expect(policyLink).toBeInTheDocument();
        });
    });

    test("renders BranchName component", () => {
        const { getByTestId } = setup();
        const branchName = getByTestId("branch-name");
        expect(branchName).toBeInTheDocument();
    });

    test("does not render social links without translations", () => {
        const { queryByTestId } = setup({
            translations: {}
        });
        footerConfig.socialLinks.forEach(link => {
            const socialLink = queryByTestId(link.key);
            expect(socialLink).not.toBeInTheDocument();
        });
    });

    test("does not render section links without translations", () => {
        const { queryByTestId } = setup({
            translations: {
                ...footerConfig.sections.reduce((acc, section) => {
                    acc[section.title] = section.title;
                    return acc;
                }, {})
            }
        });
        footerConfig.sections.forEach(section => {
            section.links.forEach(link => {
                const sectionLink = queryByTestId(link.key);
                expect(sectionLink).not.toBeInTheDocument();
            });
        });
    });

    test("renders copyright text", () => {
        const { getByTestId } = setup();
        const copyright = getByTestId("copyright_label");
        expect(copyright).toBeInTheDocument();
        expect(copyright).toHaveTextContent(
            "© 2024 Copyright Unicity International, Inc. All Rights Reserved."
        );
    });

    test("renders policy separators correctly", () => {
        const { container } = setup();
        const separators = container.querySelectorAll(".separator");
        expect(separators.length).toBe(footerConfig.policies.length - 1);
    });

    test("applies correct attributes to social links", () => {
        const { getByTestId } = setup();
        footerConfig.socialLinks.forEach(link => {
            const socialLink = getByTestId(link.key);
            expect(socialLink).toHaveAttribute("href", link.key);
            expect(socialLink).toHaveAttribute("target", "_blank");
            expect(socialLink).toHaveAttribute("rel", "noreferrer");
        });
    });

    test("applies correct attributes to section links", () => {
        const { getByTestId } = setup();
        footerConfig.sections.forEach(section => {
            section.links.forEach(link => {
                const sectionLink = getByTestId(link.key);
                expect(sectionLink).toHaveAttribute("href", link.key);
                expect(sectionLink).toHaveAttribute("target", "_blank");
                expect(sectionLink).toHaveAttribute("rel", "noreferrer");
            });
        });
    });
    describe("Footer link mixpanel tracking", () => {
        const TRACKING_TEST_CASES = [
            // Section links
            {
                testId: "footer_our_company_link",
                expectedAction: "our_company"
            },
            { testId: "footer_science_link", expectedAction: "science" },
            {
                testId: "footer_unicity_blog_link",
                expectedAction: "unicity_blog"
            },
            { testId: "footer_contact_us_link", expectedAction: "contact" },
            // Social media links
            { testId: "footer_twitter_link", expectedAction: "social_x" },
            {
                testId: "footer_facebook_link",
                expectedAction: "social_facebook"
            },
            {
                testId: "footer_pintrest_link",
                expectedAction: "social_pinterest"
            },
            {
                testId: "footer_instagram_link",
                expectedAction: "social_instagram"
            },
            { testId: "footer_youtube_link", expectedAction: "social_youtube" }
        ];

        test.each(TRACKING_TEST_CASES)(
            "tracks $testId click with action $expectedAction",
            ({ testId, expectedAction }) => {
                const { getByTestId } = setup();
                const link = getByTestId(testId);

                link.click();

                expect(mixpanelService.trackEvent).toHaveBeenCalledWith(
                    "enr_footer_link_clicked",
                    { action: expectedAction }
                );
            }
        );

        test("does not track events on render", () => {
            setup();
            expect(mixpanelService.trackEvent).not.toHaveBeenCalled();
        });

        test("tracks each link click only once", () => {
            const { getByTestId } = setup();
            const link = getByTestId("footer_our_company_link");

            link.click();

            expect(mixpanelService.trackEvent).toHaveBeenCalledTimes(1);
        });

        test("handles tracking errors gracefully", () => {
            const { getByTestId } = setup();
            const link = getByTestId("footer_our_company_link");

            mixpanelService.trackEvent.mockImplementationOnce(() => {
                throw new Error("Mixpanel error");
            });

            // Should not throw when clicking
            expect(() => link.click()).not.toThrow();
        });
    });
});
