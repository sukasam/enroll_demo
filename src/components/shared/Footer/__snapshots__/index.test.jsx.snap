// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component matches snapshot 1`] = `
<footer
  class="footer css-13by3ua"
>
  <div
    class="top-wrapper"
  >
    <div
      class="top"
    >
      <div
        class="social"
      >
        <img
          alt="test-img"
          class="logo"
          data-testid="Unicity Logo"
          height="25"
          src="/svg/logo.svg"
          width="140"
        />
        <div
          class="social-links"
        >
          <a
            class="social-link"
            data-testid="footer_facebook_link"
            href="footer_facebook_link"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="test-img"
              height="24"
              src="/svg/facebook.svg"
              width="24"
            />
          </a>
          <a
            class="social-link"
            data-testid="footer_twitter_link"
            href="footer_twitter_link"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="test-img"
              height="24"
              src="/svg/x.svg"
              width="24"
            />
          </a>
          <a
            class="social-link"
            data-testid="footer_pintrest_link"
            href="footer_pintrest_link"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="test-img"
              height="24"
              src="/svg/pintrest.svg"
              width="24"
            />
          </a>
          <a
            class="social-link"
            data-testid="footer_instagram_link"
            href="footer_instagram_link"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="test-img"
              height="24"
              src="/svg/instagram.svg"
              width="24"
            />
          </a>
          <a
            class="social-link"
            data-testid="footer_youtube_link"
            href="footer_youtube_link"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="test-img"
              height="24"
              src="/svg/youtube.svg"
              width="24"
            />
          </a>
        </div>
      </div>
      <div
        class="sections"
      >
        <div
          class="section"
        >
          <p
            class="bolder"
            data-testid="footer_about"
          >
            footer_about
          </p>
          <a
            class="links"
            data-testid="footer_our_company_link"
            href="footer_our_company_link"
            rel="noreferrer"
            target="_blank"
          >
            <div>
              footer_our_company
            </div>
          </a>
          <a
            class="links"
            data-testid="footer_science_link"
            href="footer_science_link"
            rel="noreferrer"
            target="_blank"
          >
            <div>
              footer_science
            </div>
          </a>
          <a
            class="links"
            data-testid="footer_unicity_blog_link"
            href="footer_unicity_blog_link"
            rel="noreferrer"
            target="_blank"
          >
            <div>
              footer_unicity_blog
            </div>
          </a>
        </div>
        <div
          class="section"
        >
          <p
            class="bolder"
            data-testid="footer_customer_support"
          >
            footer_customer_support
          </p>
          <a
            class="links"
            data-testid="footer_contact_us_link"
            href="footer_contact_us_link"
            rel="noreferrer"
            target="_blank"
          >
            <div>
              footer_contact_us
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    class=MuiGrid-root MuiGrid-container bottom css-XXXXX-MuiGrid-root
  >
    <div
      class="bottom-items"
    >
      <div
        class="copyright"
        data-testid="copyright_label"
      >
        Â© 2024 Copyright Unicity International, Inc. All Rights Reserved.
      </div>
      <div
        class="policy-container"
      >
        <div>
          <a
            class="policy"
            data-testid="footer_privacy_policy_link"
            href="footer_privacy_policy"
            rel="noreferrer"
            target="_blank"
          >
            footer_privacy_policy
          </a>
          <span
            class="separator"
          >
            |
          </span>
        </div>
        <div>
          <a
            class="policy"
            data-testid="footer_terms_of_use_link"
            href="footer_terms_of_use"
            rel="noreferrer"
            target="_blank"
          >
            footer_terms_of_use
          </a>
        </div>
      </div>
    </div>
    <div
      data-testid="branch-name"
    >
      BranchName
    </div>
  </div>
</footer>
`;
